sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -206,15 +207,35 @@ def _infer_event_timestamp_from_dataframe(entity_df: pandas.DataFrame) -> str:
 
 
 class BigQueryRetrievalJob(RetrievalJob):
-    def __init__(self, query, client):
+    def __init__(self, query, client, config):
         self.query = query
         self.client = client
+        self.config = config
 
     def to_df(self):
         # TODO: Ideally only start this job when the user runs "get_historical_features", not when they run to_df()
         df = self.client.query(self.query).to_dataframe(create_bqstorage_client=True)
         return df
 
+    def to_bigquery(self, dry_run=False) -> str:
+        path = f"{self.client.project}.{self.config.offline_store.dataset}.tmp-{str(uuid.uuid4())}"
```

Comment:
should we perhaps use underscores and maybe use a more descriptive prefix for the table, like `training_`. Also, we can possibly add a date for convenience and ordering. 

`training_20210614_c7927c7f`

I would also trim the UUID to just 7 or 8 characters.

Reply:
done!